src/main/groovy/me/tatarka/RetrolambdaPluginAndroid.groovy

```diff
@@ -123,7 +123,8 @@ public class RetrolambdaPluginAndroid implements Plugin<Project> {
         }
 
         JavaCompile compileUnitTest = (JavaCompile) project.tasks.find { task -> 
-            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") 
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJava") ||
+            task.name.startsWith("compile${variant.capitalize()}UnitTestJavaWithJavac")
```

Comment:
Btw, you could avoid string matching at all. My mistake...
Here should be simply project.android.unitTestVariants.all {} closure.